axum/CHANGELOG.md

```diff
@@ -266,22 +296,53 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   Becomes this in 0.6:
 
   ```rust
-  struct MyExtractor;
+  use axum::{
+      extract::{FromRequest, FromRequestParts},
+      http::{StatusCode, Request, request::Parts},
+      async_trait,
+  };
+
+  struct MyExtractor { /* ... */ }
+
+  // implement `FromRequestParts` if you don't need to consume the request body
+  #[async_trait]
+  impl<S> FromRequestParts<S> for MyExtractor
+  where
+      S: Send + Sync,
+      B: Send + 'static,
+  {
+      type Rejection = StatusCode;
+  
+      async fn from_request_parts(parts: &mut Parts, state: &S) -> Result<Self, Self::Rejection> {
+          // ...
+      }
+  }
 
+  // implement `FromRequest` if you do need to consume the request body
   #[async_trait]
   impl<S, B> FromRequest<S, B> for MyExtractor
   where
       S: Send + Sync,
-      B: Send,
+      B: Send + 'static,
   {
       type Rejection = StatusCode;
   
-      async fn from_request(req: &mut RequestParts<S, B>) -> Result<Self, Self::Rejection> {
+      async fn from_request(req: Request<B>, state: &S) -> Result<Self, Self::Rejection> {
           // ...
       }
   }
   ```
 
+- **breaking:** `RequestParts` has been removed as part of the `FromRequest`
+  rework ([#1272])
+- **breaking:** `BodyAlreadyExtracted` has been removed ([#1272])
+- **breaking:** The following types or traits have a new `S` type param
+  (`()` by default) which represents the state ([#1155]):
+  - `FromRequest`
```

Comment:
This isn't right, `FromRequest` doesn't have the `()` default. `Handler` does I guess, but should it actually?

Reply:
> `Handler` does I guess, but should it actually?

Hm good question 🤔 The main reason for the defaults in the first place was to make `impl Handler` easier to use. But I don't think thats possible anymore since the `M` types are private. I don't mind that actually, `State` and closure captures is easier to use anyway.

So I've removed the default for `Handler` since we probably don't need it.